.github/workflows/count_changed_files.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/env python3
+# -*- coding: UTF-8 -*-
+"""Script to limit number of file changes in single PR.
+
+Methodology:
+
+    Analyses the Pull request to find if the count of file changed in a PR
+    exceeds a pre-defined nummber 20
+
+    This scripts encourages contributors to align with project practices,
+    reducing the likelihood of unintentional merges into incorrect branches.
+
+NOTE:
+
+    This script complies with our python3 coding and documentation standards.
+    It complies with:
+
+        1) Pylint
+        2) Pydocstyle
+        3) Pycodestyle
+        4) Flake8
+
+"""
+
+import sys
+import argparse
+import subprocess
+
+
+def _count_changed_files(base_branch, PR_branch):
+    """
+    Count the number of changed files between two branches.
+
+    Args:
+        base_branch (str): The base branch.
+        PR_branch (str): The PR branch.
+
+    Returns:
+        int: The number of changed files.
+
+    Raises:
+        SystemExit: If an error occurs during execution.
+    """
+    base_branch = f"origin/{base_branch}"
+    PR_branch = f"origin/{PR_branch}"
+    try:
+        # Run git command to get the list of changed files
+        command = f"git diff --name-only {base_branch}...{PR_branch} | wc -l"
+        process = subprocess.Popen(
+            command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        output, error = process.communicate()
```

Comment:
Only have this statement in the `try` block. It is the only one where success is not guaranteed.

Reply:
When you have more than one statement in the block, troubleshooting becomes harder